bin/zbrewmount

```diff
@@ -0,0 +1,185 @@
+            ;;
+        t )
+            opt_combine_repeat_chk -st
+            tgtmode="true"
+            ;;
+        v )
+            verbose="true"
+            ;;
+        \?)
+            if [ ${OPTARG} != "?" ]; then
+                echo "Invalid option specified: -""${OPTARG}"
+            fi
+            syntax
+            ;;
+     esac
+done
+shift $(expr $OPTIND - 1 )
+
+if [ -z "$mount" ] && [ -z "$unmount" ]; then
+    echo "** ERROR: must specify mount -m or unmount -u option" >&2
+    syntax
+fi
+
+if [ -z "$srcmode" ] && [ -z "$tgtmode" ]; then
+    echo "** ERROR: must specify -s or -t option" >&2
+    syntax
+fi
+
+sw=$1
+if [ -z $sw ]; then
+    echo "**ERROR: sw name must be specified" >&2
+    syntax
+fi
+shift
+ussname=$(echo ${sw} | tr '[:upper:]' '[:lower:]');
+zosname=$(echo ${sw} | tr '[:lower:]' '[:upper:]');
+prefix=`echo "${ussname}" | awk '{ print substr($1, 0, 3) }'`
+bom_file="${ZBREW_REPOROOT}/zbrew-${prefix}/${ussname}/${ussname}bom.json"
+}
+
+function run_unmount {
+    mountpoints=$1
+    zbrew_zfsroot=$2
+    zbrew_hlq=$3
+    zosname=$4
+
+    for mount in $(echo $mountpoints | sed "s/,/ /g"); do
+        absmount="${zbrew_zfsroot}${mount#*:}"
+        if [ ! -z $verbose ]; then
+            echo "**INFO: Attempting to unmount $absmount"
+        fi
+        if [ -d "${absmount}" ]; then
+            /usr/sbin/unmount -R "${absmount}" 2>/dev/null
+        fi
+    done
+    
+    if [ ! -z $verbose ]; then
+        echo "**INFO: Attempting to remove ${zbrew_hlq}${zosname}*.**"
+    fi
+    drm -f "${zbrew_hlq}${zosname}*.**"
+}
+
+function run_mount {
+    mountpoints=$1
+    zbrew_zfsroot=$2
+    zbrew_hlq=$3
+
+    for mount in $(echo $mountpoints | sed "s/,/ /g"); do
+        relds="${zosname}.${mount%%:*}"	    #${mount%%:*} strips everything after first occurance of ':' including ':'
+        relzfs="${mount#*:}"    #${mount#*:} strips everything before first occurance of ':' including ':'
+        absmount="${zbrew_zfsroot}${relzfs}"
+
+        mkdir -p -m 755 "${absmount}"
+        chk $? "**ERROR: making directory ${absmount}"
+
+        if [ ! -z $verbose ]; then
+            echo "**INFO: Attempting to mount $absmount"
+        fi
+        
+        /usr/sbin/mount -t zfs -f "${zbrew_hlq}${relds}" "${absmount}"
+        chk $? "**ERROR: unable to mount directory: ${absmount} at ${zbrew_hlq}${relds}"
+    done
+}
+
+
+# ---------------------------------------------------------------------
+# --- main
+# ---------------------------------------------------------------------
+. zbrewsetenv
+
+read_args $*
+
+if ! [ -f "${bom_file}" ]; then
+	echo "**ERROR: unable to find BOM file: ${bom_file}" >&2
+	exit 8
+fi
+
+readbom_out=`readbom $ussname <$bom_file`
+chk $? "**ERROR: readbom failed"
+
+mountpoints=`echo "${readbom_out}" | awk '($2 == "ZFS") {print $1 ":" $7}'`
+mountpoints=$(echo $mountpoints | tr ' ' ',')
+
+# different scenarios
+if [ ! -z $mount ] && [ ! -z $srcmode ]; then
```

Comment:
it's up to you, but i find it easier to understand multiple conditions like this if you do it in 2 layers, e.g.
if <mount> then
  if <src> then
    ..
  else 
    ...
  end
else # unmount
   if <src> then
     ..
   else 
    ..
  end
end